mailet/standard/src/main/java/org/apache/james/transport/matchers/AtMost.java

```diff
@@ -33,39 +35,80 @@
+ * <p> In the mail, every time the check succeeds, its counter will be incremented by one.
+ * The check fails when the defined X limit is reached.</p>
  *
- * The check fails when the defined X limit is reached.
+ * <ul>
+ * <li>X - count of how many times a specific operation is performed</li>
+ * <li>Y - name of attribute represented for specific operation executions, default value is: <i>AT_MOST_EXECUTIONS</i></li>
+ * </ul>
  *
  * <p>The example below will match a mail with at most 3 executions on the mailet</p>
+ * with attribute name <i>AT_MOST_EXECUTIONS</i></p>
  *
  * <pre><code>
- * &lt;mailet match=&quot;AtMost=3&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * &lt;mailet match=&quot;AtMost=AT_MOST_EXECUTIONS:3&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
  * &lt;/mailet&gt;
  * </code></pre>
  */
 public class AtMost extends GenericMatcher {
     static final AttributeName AT_MOST_EXECUTIONS = AttributeName.of("AT_MOST_EXECUTIONS");
+    private static final String CONDITION_SEPARATOR = ":";
+    private static final int ONLY_CONDITION_VALUE = 1;
+    private static final int CONDITION_NAME_AND_VALUE = 2;
+
+    private AttributeName attributeName;
     private Integer atMostExecutions;
 
     @Override
     public void init() throws MessagingException {
-        this.atMostExecutions = MailetUtil.getInitParameterAsStrictlyPositiveInteger(getCondition());
+        String conditionConfig = getMatcherConfig().getCondition();
+        Preconditions.checkArgument(StringUtils.isNotBlank(conditionConfig), "MatcherConfiguration is mandatory!");
+        Preconditions.checkArgument(!conditionConfig.startsWith(CONDITION_SEPARATOR),
+            "MatcherConfiguration can not start with '%s'", CONDITION_SEPARATOR);
+
+        List<String> conditions = Splitter.on(CONDITION_SEPARATOR).splitToList(conditionConfig);
+        attributeName = parseAttribute(conditions);
+        atMostExecutions = parseAttributeValue(conditions);
+    }
+
+    private AttributeName parseAttribute(List<String> conditions) {
+        switch (conditions.size()) {
+            case ONLY_CONDITION_VALUE:
+                return AT_MOST_EXECUTIONS;
+            case CONDITION_NAME_AND_VALUE:
+                return AttributeName.of(conditions.get(0));
+            default:
+                throw new IllegalArgumentException("MatcherConfiguration format should follow: 'name:value'");
+        }
+    }
+
+    private Integer parseAttributeValue(List<String> conditions) throws MessagingException {
+        switch (conditions.size()) {
+            case ONLY_CONDITION_VALUE:
+                return MailetUtil.getInitParameterAsStrictlyPositiveInteger(conditions.get(0));
+            case CONDITION_NAME_AND_VALUE:
+                return MailetUtil.getInitParameterAsStrictlyPositiveInteger(conditions.get(1));
+            default:
+                throw new IllegalArgumentException("MatcherConfiguration format should follow: 'name:value'");
```

Comment:
idem? maybe precise that value should be a strictly positive integer?

Reply:
The error message is handled as part of MailetUtil